capstone-rs/src/capstone.rs

```diff
@@ -165,6 +165,37 @@ impl Capstone {
         }
     }
 
+    pub fn malloc<'a>(&'a self) -> *mut cs_insn {
+        unsafe { cs_malloc(self.csh()) }
+    }
+
+    pub fn free<'a>(&'a self, insn: *mut cs_insn) {
+        unsafe {
+            cs_free(insn, 1);
+        }
+    }
+
+    pub fn disasm_iter<'a>(
```

Comment:
I need to think more about what the `disasm_iter()` function interface should be.

Instead of a C-like interface, we want a "Rusty" interface that:

1. Does not require the user to use unsafe
    a. Does not take any pointers
2. Does not require manual malloc/free functions to avoid memory leaks, double frees, etc.
3. Is easy to use
    a. Does not return a tuple (an enum/struct instead)

Reply:
I agree. Let me close this PR. I can do a followup with these guidelines